core/src/main/java/io/fabric8/maven/core/config/MappingConfig.java

```diff
@@ -0,0 +1,28 @@
+package io.fabric8.maven.core.config;
+
+public class MappingConfig {
+
+    private String kind;
+
+    private String filenameTypes;
+
+    public String getKind() {
+        return kind;
+    }
+
+    public String getFilenameTypes() {
+        return filenameTypes;
+    }
+
+    public String[] getFilenamesAsArray() {
+        if (this.filenameTypes == null) {
+            return new String[0];
+        }
+        return filenameTypes.split(",\\s*");
+    }
+
+    public boolean isValid() {
+        return kind != null &&  filenameTypes.length() > 0;
```

Comment:
Final remark, promised ;):

I think this can still result in a NPE. Better getFilenamesAsArray() > 0.

BTW, I think it would be better to throw an error if an invalid mapping is given instead of just ignoring it. So maybe rename this to `void validate()` and thrown an `IllegalArgumentException` if invalid ?

Reply:
Oh sorry true.